### DISCUSSION TITLE
Standings for particular week

### DISCUSSION THREAD

**Discussion by brthurr** [cwendt94/espn-api#511] [Q&A]
I think I saw this asked and answered via a PR that didn't get committed, so my apologies if this is the case.

Is there a way to get the football league standings for a given week? I'm playing around with a custom losers bracket and I'd like to populate my database with historical values for testing. league.standings() only gives the most recent standings. I'd like to be able to see, for example, the standings from week 14 of 2022.

**Comment by cwendt94** [comment]
Thats a great question, Im not sure if this is possible or not. I need to check if there is a way to get the teams record for a specific scoring period/week

**Comment by brthurr** [comment]
I actually used your power_rankings as a baseline, which does accept a week argument.

```
def standings_weekly(self, week: int = None) -> List[Team]:
        if not week or week <= 0 or week > self.current_week:
            week = self.current_week

        # Initialize a dictionary to hold team records
        team_records = {team.team_id: {'wins': 0, 'losses': 0, 'points_for': 0, 'points_against': 0} for team in self.teams}

        # Calculate wins, losses, points for, and points against for each team
        for team in self.teams:
            for i in range(week):
                opponent = team.schedule[i]
                team_points = team.scores[i]
                opponent_points = opponent.scores[i]
                if team_points > opponent_points:
                    team_records[team.team_id]['wins'] += 1
                else:
                    team_records[team.team_id]['losses'] += 1
                team_records[team.team_id]['points_for'] += team_points
                team_records[team.team_id]['points_against'] += opponent_points

        # Sort teams based on wins, then points for, and then points against
        sorted_teams = sorted(self.teams, key=lambda x: (team_records[x.team_id]['wins'], team_records[x.team_id]['losses'], team_records[x.team_id]['points_for'], -team_records[x.team_id]['points_against']), reverse=True)

        return sorted_teams
```

This could be used as a standalone method, or as an enhancement to standings. Thoughts?

I'd be happy to create a PR if you like the idea.

**Reply by cwendt94** [reply]
Yeah, I think this is a good idea and having as a standalone method!
